mlflow/sagemaker/__init__.py

```diff
@@ -380,8 +380,10 @@ def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=No
                          operation_status.message)
         else:
             raise MlflowException(
-                "The deployment operation failed with the following error message:"
-                " \"{error_message}\"".format(error_message=operation_status.message))
+                message=(
+                    "The deployment operation failed with the following error message:"
+                    " \"{error_message}\"".format(error_message=operation_status.message)),
+                error_code=PUBLIC)
```

Comment:
is `PUBLIC` a type of error code? Looks like it is API visibility type

Reply:
Good call - removed `PUBLIC` and reverted back to `INTERNAL_ERROR`, since everything else seemed misleading.